test/browser/call-search.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+import { PeerUser } from '../peer-user.js';
+import { expectEvents, updateRemoteVideos, sleep, logEvents } from '../helper.js';
+import config from './config.js'
+
+const assert = chai.assert;
+let client;
+let peerUser;
+let call;
+describe('Call Search', async function() {
+    this.timeout(300000);
+
+    before(async function() {
+        Circuit.logger.setLevel(Circuit.Enums.LogLevel.Error);
+        client = new Circuit.Client(config.config);
+        const res = await Promise.all([PeerUser.create(), client.logon(config.credentials)]);
+        peerUser = res[0];
+        const conversation = await client.createGroupConversation([peerUser.userId], 'SDK Test: Conference Call');
+        call = await client.startConference(conversation.convId, {audio: true, video: false});
+        await expectEvents(client, [{
+            type: 'callStatus',
+            predicate: evt => evt.call.state === Circuit.Enums.CallStateName.Initiated
+        }, {
+            type: 'callStatus',
+            predicate: evt => evt.call.state === Circuit.Enums.CallStateName.Waiting
+        }]);
+        await sleep(3000); // wait to make sure the call is ready to be joined
+        await Promise.all([
+            peerUser.exec('joinConference', call.callId, {audio: true, video: false}),
+            expectEvents(client, [{
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'callStateChanged' && evt.call.state === Circuit.Enums.CallStateName.Active
+            }, {
+                type: 'callStatus',
+                predicate: evt => evt.reason === 'participantJoined'
+            }])
+        ]);
+        call = await client.findCall(call.callId);
+    });
+
+    after(async function() {
+        await Promise.all([peerUser.destroy(), client.logout()]);
+    });
+
+    afterEach(async function() {
+        client.removeAllListeners();
+    });
+
+    it('should get the active call', async () => {
+        const res = await client.getActiveCall();
+        assert(res.callId === call.callId);
+    });
+
+    it('should get all calls and verify it contains the active call', async () => {
+        const res = await client.getCalls();
+        assert(res && res.some(c => c.callId === call.callId));
+    });
+
+    it('find a call by its id', async () => {
```

Comment:
A general question: Do you think it would be better to name all the tests by its API that is being tested rather than what we have?

E.g. have:
it('function findCall', async () => { ...

rather than:
it('find a call by its id', async () => {

IMO, the API name might be better since we actually test an API. And it would also be clearer as to what API fails.

Reply:
I agree, I can do that.